### ISSUE TITLE
A wrong behavior after upgrade to 4.14.0

### ISSUE THREAD

**Issue by mlc-mlapis** [typescript-eslint/typescript-eslint#2947]
- [x] I have tried restarting my IDE, and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

If `AppState` is imported:
```TS
import {AppState} from '@m-kbase/shared/general/states';
```
and later used **only** in a constructor, like:
```TS
constructor(
   private _store: Store<AppState>
) {}
```
then the warning appears in many places (in relation to the `import {AppState} from '@m-kbase/shared/general/states';` line):

`warning  'AppState' is defined but never used  @typescript-eslint/no-unused-vars`

**Expected Result**

The same result as with the previous version 4.13.0 or earlier, with no warnings.

**Additional Info**

Probably similar to https://github.com/typescript-eslint/typescript-eslint/issues/2946

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.14.0` |
| `@typescript-eslint/parser`        | `4.14.0` |
| `TypeScript`                       | `4.0.5` |
| `ESLint`                           | `7.18.0` |
| `node`                             | `12.13.1` |

**Comment by chunghha** [comment]
I see same warning after updating from 4.13.0 to 4.14.0. For example with this code, it throws "**3:22  warning  'HealthResponse' is defined but never used  @typescript-eslint/no-unused-vars**",

```
import { Controller, Get } from '@nestjs/common';

import { AppService, HealthResponse } from './app.service';

@Controller('tfapi')
export class AppController {
	constructor(private readonly appService: AppService) {}

	@Get('/isHealthy')
	async getHealth(): Promise<HealthResponse> {
		return this.appService.getHealth();
	}
}

```

**Comment by vladimiry** [comment]
Same here. Btw, the warning doesn't get triggered in the case of Inline type import use:
```
constructor(
   private _store: Store<import("@m-kbase/shared/general/states").AppState>,
) {}
```

**Comment by bradzacher** [comment]
Same as #2941, #2942, #2945

Fixed in https://github.com/typescript-eslint/typescript-eslint/pull/2943

Please use the search next time.

**Comment by mlc-mlapis** [comment]
@bradzacher Thanks for your PR, I was too fast. You are right. 🥇
